packages/coil-extension/src/content/services/ContentScript.ts

```diff
@@ -156,7 +174,38 @@ export class ContentScript {
   }
 
   init() {
-    if (this.frames.isTopFrame) {
+    if (this.frames.isMonetizableFrame) {
+      this.frames.monitor()
+    }
+
+    // TODO: we could allow arbitrary frames
+    if (this.frames.isDirectChildFrame) {
+      this.window.addEventListener('message', event => {
+        const data = event.data
+        if (typeof data.wmIframe === 'object') {
+          const {
+            allowToken,
+            allowed
+          }: { allowToken: string; allowed: boolean } = data.wmIframe
+          if (allowToken === this.allowToken) {
```

Comment:
I guess potentially this could be listened in on, then the message could be sent again with a different `allow`. 
We could clear `this.allowToken`